test/lib/jdk/test/lib/apps/LingeredApp.java

```diff
@@ -441,15 +444,44 @@ public void stopApp() throws IOException {
      * @throws IOException
      */
     public static void startAppExactJvmOpts(LingeredApp theApp, String... jvmOpts) throws IOException {
+        long t1 = System.currentTimeMillis();
         theApp.createLock();
         try {
             theApp.runAppExactJvmOpts(jvmOpts);
-            theApp.waitAppReady();
+            theApp.waitAppReadyOrCrashed();
         } catch (Exception ex) {
-            System.out.println("LingeredApp failed to start: " + ex);
-            theApp.finishApp();
+            boolean alive = theApp.getProcess() != null && theApp.getProcess().isAlive();
+            System.out.println("LingeredApp failed to start or failed to crash. isAlive=" + alive + ": " + ex);
+            // stopApp in case it is still alive, may be able to get output:
+            if (alive) {
+                theApp.stopApp();
+            }
+            alive = theApp.getProcess() != null && theApp.getProcess().isAlive();
+            if (!alive) {
+                theApp.finishApp(); // Calls getOutput(), fails if still alive
+            }
             theApp.deleteLock();
             throw ex;
+        } finally {
+            long t2 = System.currentTimeMillis();
+            System.out.println("LingeredApp startup took " + (t2 - t1) + "ms");
+            checkForDumps();
+        }
+    }
+
+    /**
+      * Show any dump files of interest in the current directory.
+      */
+    public static void checkForDumps() {
+        System.out.println("Check for hs_err_pid/core/mdmp files:");
+        int count = 0;
+        FilenameFilter filter = (dir, file) -> (file.startsWith("hs_err_pid") || file.startsWith("core") || file.endsWith("mdmp"));
+        for (File f : new File(".").listFiles(filter)) {
+            System.out.println(f);
```

Comment:
Printing the size can be useful for core files.

Reply:
Yes can add that.  It could be useful as occasionally we wonder if a core/mdmp is slow to write.  A file size at this point might be useful to compare with a file size viewed later.